src/fast_check/transform.rs

```diff
@@ -903,6 +921,101 @@ impl<'a> FastCheckTransformer<'a> {
     Ok(())
   }
 
+  fn transform_arrow(
+    &mut self,
+    n: &mut ArrowExpr,
+    parent_id_range: Option<SourceRange>,
+  ) -> Result<(), Box<FastCheckDiagnostic>> {
+    let range = parent_id_range.unwrap_or_else(|| n.range());
+
+    let mut clear_body = true;
+
+    if n.return_type.is_none() {
+      if n.is_generator {
+        self.mark_diagnostic(
+          FastCheckDiagnostic::MissingExplicitReturnType {
+            range: self.source_range_to_range(range),
+          },
+        )?;
+      }
+
+      match &mut *n.body {
+        BlockStmtOrExpr::BlockStmt(body) => {
+          let return_stmts = get_return_stmts_with_arg_from_function_body(body);
+          if return_stmts.is_empty() {
+            let void_type =
+              Box::new(ts_keyword_type(TsKeywordTypeKind::TsVoidKeyword));
+            n.return_type = Some(Box::new(TsTypeAnn {
+              span: DUMMY_SP,
+              type_ann: if n.is_async {
+                Box::new(TsType::TsTypeRef(TsTypeRef {
+                  span: DUMMY_SP,
+                  type_name: TsEntityName::Ident(Ident::new(
+                    "Promise".into(),
+                    DUMMY_SP,
+                  )),
+                  type_params: Some(Box::new(TsTypeParamInstantiation {
+                    span: DUMMY_SP,
+                    params: vec![void_type],
+                  })),
+                }))
+              } else {
+                void_type
+              },
+            }));
```

Comment:
This code seems the same as above? Maybe we could add some helper functions at the bottom of this file for more easily creating these `TsTypeAnn`s.

Reply:
de-duped